### PR TITLE
[Javascript] Manual Instrumentation for cache module

### PR DESCRIPTION
**Pull request by DominikB2014**
This is the javascript version of these docs https://github.com/getsentry/sentry-docs/pull/9926
Also updates some incorrect cache module docs links

### CODE REVIEW

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+To make it possible for Sentry to give you an overview of your cache performance, you'll need to create two spans - one indicating that something is being put into the cache, and a second one indicating that something is being fetched from the cache.
+
+Make sure that there's a transaction running when you create the spans. If you're using a web framework those transactions will be created for you automatically. See <PlatformLink to="/performance/">Performance Monitoring</PlatformLink> for more information.
+
+For detailed information about which data can be set, see the [Cache Module Developer Specification](https://develop.sentry.dev/sdk/performance/modules/caches/).
+
+## Manual Instrumentation
+
+If you're using anything other than Sentrys redis integration, you'll need to manually instrument the [Cache Module](<(https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/)>) by following the steps below.
```
Comment (DominikB2014) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:21]: Maybe not a block for this PR, but we should probably link to the sentry Redis integration for sentry, so we can show how to do 
```
integrations: [Sentry.redisIntegration({ cachePrefixes: ['cache:'] })]
```

However, it looks like we don't even have a page dedicated to this integration, maybe here?
https://docs.sentry.io/platforms/javascript/guides/node/performance/database/#supported-platforms

**Comment by DominikB2014** [comment]
> IMHO this page should either be at the same level as "Custom Instrumentation for Requests", or we should nest both into the same level? depending on how many more of these were are planning to add?

@mydea im in favour of nesting them all under custom instrumentation. This would match what we do in python too!

**Comment by DominikB2014** [comment]
@mydea I've moved the http docs, and matched the title to be like pythons for consistency. I couldn't find any links to the old location making a redirect uncessary, but I tagged @gggritso who wrote the http js docs, who can see if they know of any links.

**Review by gggritso — APPROVED** [review]

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx (hunk 1 above)
Comment (DominikB2014) [reply docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:21]: 👍

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
+---
+title: Instrument Caches
+sidebar_order: 1000
+description: "Learn how to manually instrument your code to use Sentrys Cache module. "
```
Comment (lizokm) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:4]:
```suggestion
description: "Learn how to manually instrument your code to use Sentry's Cache module. "
```

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx (hunk 1 above)
Comment (lizokm) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:21]:
```suggestion

## Manual Instrumentation

If you're using anything other than Sentry's redis integration, you'll need to manually instrument the [Cache Module](<(https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/)>) by following the steps below.

You'll need to create two spans - one indicating that something is being put into the cache, and a second one indicating that something is being fetched from the cache.

Make sure that there's a transaction running when you create the spans. If you're using a web framework those transactions will be created for you automatically. See <PlatformLink to="/performance/">Performance Monitoring</PlatformLink> for more information.

For detailed information about which data can be set, see the [Cache Module Developer Specification](https://develop.sentry.dev/sdk/performance/modules/caches/).



```

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+If the cache you’re using isn’t supported by auto instrumentation mentioned above, you can use the custom instrumentation instructions below to emit cache spans:
+
+1. Set the cache value with whatever cache library you happen to be using.
+2. Wrap the part of your application that uses the cached value with `Sentry.startSpan(...)`.
+3. Set the `name` to something descriptive like "Setting auth cache".
+4. Set `op` to `cache.set`.
+5. Set `cache.key` to a string array representing the key(s) you are setting.
+6. Other attributes such as `cache.item_size` can be optionally set, see [Cache Module Span Data Conventions](https://develop.sentry.dev/sdk/performance/modules/caches/#span-data) for more information.
+
+(The steps described above are documented in the snippet.)
```
Comment (lizokm) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:38]:
```suggestion
If the cache you’re using isn’t supported by the auto instrumentation mentioned above, you can use the custom instrumentation instructions below to emit cache spans instead:

1. Set the cache value with whatever cache library you happen to be using.
2. Wrap the part of your application that uses the cached value with `Sentry.startSpan(...)`.
3. Set the `name` to something descriptive like "Setting auth cache".
4. Set `op` to `cache.set`.
5. Set `cache.key` to a string array representing the key(s) you're setting.
6. Optionally, you can set other attributes such as `cache.item_size`. (See [Cache Module Span Data Conventions](https://develop.sentry.dev/sdk/performance/modules/caches/#span-data) for more information.)

(The steps described above are also documented in the snippet.)
```

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  (span) => {
+    // Set a key in your caching using your custom caching solution
+    my_caching.set(key, value);
+  }
+);
+```
+
+### Add Span When Retrieving Data From the Cache
+
+If the cache you’re using isn’t supported by auto instrumentation mentioned above, you can use the custom instrumentation instructions below to emit cache spans:
```
Comment (lizokm) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:64]:
```suggestion
If the cache you’re using isn’t supported by the auto instrumentation mentioned above, you can use the custom instrumentation instructions below to emit cache spans instead:
```

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+1. Get the cached value from whatever cache library you happen to be using.
+2. Wrap the part of your application that fetches from the cache with `Sentry.startSpan(...)`.
+3. Set the `name` to something descriptive like "Getting auth cache".
+4. Set `op` to `cache.get`.
+5. Set `cache.key` to a string array representing the key(s) you are setting.
+6. Set `cache.hit` to a boolean value representing whether the value was successfully fetched from the cache or not.
+7. Other attributes such as `cache.item_size` can be optionally set, see [Cache Module Span Data Conventions](https://develop.sentry.dev/sdk/performance/modules/caches/#span-data) for more information.
+   (The steps described above are documented in the snippet.)
```
Comment (lizokm) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:73]:
```suggestion

1. Get the cached value from whatever cache library you happen to be using.
2. Wrap the part of your application that fetches from the cache with `Sentry.startSpan(...)`.
3. Set the `name` to something descriptive like "Getting auth cache".
4. Set `op` to `cache.get`.
5. Set `cache.key` to a string array representing the key(s) you're setting.
6. Set `cache.hit` to a boolean value representing whether the value was successfully fetched from the cache or not.
7. Optionally, you can set other attributes such as `cache.item_size`. (See [Cache Module Span Data Conventions](https://develop.sentry.dev/sdk/performance/modules/caches/#span-data) for more information.)
   (The steps described above are also documented in the snippet.)
```

File: docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    const cacheHit = Boolean(value);
+    if (cacheHit) {
+      span.setAttribute("cache.item_size", JSON.stringify(value).length, // Warning: if value is very big this could use lots of memory);
+    }
+    span.setAttribute("cache.hit", cacheHit);
+  }
+);
+```
+
+You should now have the right spans in place. Head over to [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/) to see how your cache is performing.
```
Comment (lizokm) [inline docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation/cache-module.mdx:100]:
```suggestion
You should now have the right spans in place. Head over to the [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/) to see how your cache is performing.
```

File: docs/platforms/python/performance/instrumentation/custom-instrumentation/caches-module.mdx [hunk 7]
```diff
@@ -98,4 +98,4 @@ with sentry_sdk.start_span(op="cache.get_item") as span:
         span.set_data("cache.hit", False)
 ```
 
-You should now have the right spans in place. Head over to [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/performance/cache/) to see how your cache is performing.
+You should now have the right spans in place. Head over to [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/) to see how your cache is performing.
```
Comment (lizokm) [inline docs/platforms/python/performance/instrumentation/custom-instrumentation/caches-module.mdx:101]:
```suggestion
You should now have the right spans in place. Head over to the [Cache dashboard](https://sentry.io/orgredirect/organizations/:orgslug/performance/caches/) to see how your cache is performing.
```

**Review by lizokm — APPROVED** [review]
Looks great, just a few small tweaks.

**Pull request merged**